pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -240,6 +241,17 @@ private TableDataManager createTableDataManager(String tableNameWithType, TableC
   @Override
   public void deleteTable(String tableNameWithType)
       throws Exception {
+
+    _tableDataManagerMap.compute(tableNameWithType, (k, v) -> {
```

Comment:
I feel it might just work to directly shut down the `TableDataManager` here without waiting for EV to converge. Can you verify if it works?

Reply:
After a second thought, it might not work because we need to ensure there is no query on this table, or it can crash the server